pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -277,58 +326,97 @@ contract ManagedPool is ManagedPoolSettings {
         uint256 balanceTokenIn,
         uint256 balanceTokenOut,
         bytes32 poolState
-    ) internal view returns (uint256) {
-        uint256 tokenInWeight;
-        uint256 tokenOutWeight;
-        uint256 scalingFactorTokenIn;
-        uint256 scalingFactorTokenOut;
-        uint256 swapFeeComplement;
-        {
-            uint256 weightChangeProgress = ManagedPoolStorageLib.getGradualWeightChangeProgress(poolState);
-            (tokenInWeight, scalingFactorTokenIn) = _getTokenInfo(request.tokenIn, weightChangeProgress);
-            (tokenOutWeight, scalingFactorTokenOut) = _getTokenInfo(request.tokenOut, weightChangeProgress);
+    ) internal view returns (uint256 amountCalculated) {
+        SwapData memory swapData = _getSwapData(request, poolState);
+        uint256 virtualSupply = getActualSupply();
```

Comment:
I think we agreed to use the virtual rather than actual supply here as it's not worth the extra gas and will affect bpt prices minimally.

Reply:
I remember asking about this and at one time it was actual, but will change it if that's the consensus.